lib/commands/init.ts

```diff
@@ -0,0 +1,114 @@
+		
+	public execute(args: string[]): IFuture<void> {
+		return (() => {
+			let projectData: any = { };
+			
+			if(this.$fs.exists(this.projectFilePath).wait()) {
+				projectData = this.$fs.readJson(this.projectFilePath).wait();
+			}
+			
+			let projectDataBackup = _.extend({}, projectData);			
+			
+			if(!projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE]) {
+				projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE] = { };
+				this.$fs.writeJson(this.projectFilePath, projectData).wait(); // We need to create package.json file here in order to prevent "No project found at or above and neither was a --path specified." when resolving platformsData				
+			}
+			
+			try {
+				
+				projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE]["id"] = this.getProjectId().wait();
+				
+				let $platformsData = this.$injector.resolve("platformsData");
+				_.each($platformsData.platformsNames, platform => {
+					let platformData: IPlatformData = $platformsData.getPlatformData(platform);
+					if(!platformData.targetedOS || (platformData.targetedOS && _.contains(platformData.targetedOS, process.platform))) {
+						projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE][platformData.frameworkPackageName] = this.getVersionData(platformData.frameworkPackageName).wait();
+					}
+				});
+				
+				this.$fs.writeJson(this.projectFilePath, projectData).wait();
+			} catch(err) {
+				this.$fs.writeJson(this.projectFilePath, projectDataBackup).wait();
+				throw err;
+			}
+			
+			this.$logger.out("Project successfully initialized."); 
+		}).future<void>()();
+	}
+	
+	public canExecute(args: string[]): IFuture<boolean> {
+		 return Future.fromResult(true);
+	}
+	
+	private get projectFilePath(): string {
+		if(!this._projectFilePath) {
+			let projectPath = path.resolve(this.$options.path || ".");
+			this._projectFilePath = path.join(projectPath, constants.PACKAGE_JSON_FILE_NAME);
+		}
+		
+		return this._projectFilePath;
+	}
+	
+	private set projectFilePath(newProjectFilePath: string) {
+		this._projectFilePath = newProjectFilePath;
+	}
+	
+	private getProjectId(): IFuture<string> {
+		return (() => {
+			let defaultAppId = this.$projectHelper.generateDefaultAppId(path.basename(path.dirname(this.projectFilePath)), constants.DEFAULT_APP_IDENTIFIER_PREFIX);
+			if(this.canUseDefaultValue) {
+				return defaultAppId;
+			}
+			
+			return this.$prompter.getString("Id:", () => defaultAppId).wait();
+		}).future<string>()();
+	}
+	
+	private getVersionData(packageName: string): IFuture<IStringDictionary> {
+		return (() => {
+			let latestVersion = this.$npmInstallationManager.getLatestVersion(packageName).wait();
+			if(this.canUseDefaultValue) {
+				return this.buildVersionData(latestVersion);
+			}
+			
+			let data = this.$npm.view(packageName, "versions").wait();
+			let versions = data[latestVersion].versions;
+			let sortedVersions = versions.sort(helpers.versionCompare).reverse();
```

Comment:
My suggestion is to define it in the platformData and pass it as parameter to this method. This way we'll set it to the minimum verified version that we are able to work with.